pkg/sitewise/auth.go

```diff
@@ -0,0 +1,114 @@
+package sitewise
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/grafana/iot-sitewise-datasource/pkg/models"
+)
+
+type EdgeAuthenticator struct {
+	Settings models.AWSSiteWiseDataSourceSetting
+}
+
+func (a *EdgeAuthenticator) Authorize(ctx context.Context) (models.AuthInfo, error) {
```

Comment:
nit: I assume `a` means `authenticator`, but it might seem cryptic out of context.